### ISSUE TITLE
False semicolon-termination error

### ISSUE THREAD

**Issue by alsorokin** [tsqllint/tsqllint#286]
The following SQL file produces an error, whereas it shouldn't:

```SQL
CREATE FUNCTION Foo.Bar ()
RETURNS TABLE

AS

RETURN
(
	WITH CTE AS
	(
		SELECT a, b FROM Foo.Baz
	)

	SELECT a, b FROM CTE -- should not trigger an error
);

GO 
```

**Comment by alsorokin** [comment]
Proposed fix is to visit all children of `CREATE FUNCTION` statement and find `RETURN` statement. Then just disable the rule for all `SELECT` statements inside the `RETURN` statement.

**Comment by nathan-boyd** [comment]
@alsorokin this is resolved correct?

**Comment by alsorokin** [comment]
@nathan-boyd I don't think so. I did report a similar issue, but it's not the same one:
https://github.com/tsqllint/tsqllint/issues/284
